### PR TITLE
[Resources.OperatingSystem]  Fix issue with detecting MacOS

### PR DESCRIPTION
**Pull request by ysolomchenko**
Fixes # Fix issue with detecting MacOS

## Changes

 Fix issue with detecting MacOS

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -496,6 +496,7 @@ jobs:
     with:
       project-name: Component[OpenTelemetry.Resources.OperatingSystem]
       code-cov-name: Resources.OperatingSystem
+      os-list: '[ "windows-latest", "ubuntu-latest", "macos-latest" ]'
```
Comment (pellared) [inline .github/workflows/ci.yml:499]: You can try this https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1964#issuecomment-2234020323

File: src/OpenTelemetry.Resources.OperatingSystem/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+## 0.1.0-alpha.2
+
+Released 2024-Jul-17
+
+* Fix issue with detecting MacOS
```
Comment (pellared) [inline src/OpenTelemetry.Resources.OperatingSystem/CHANGELOG.md:9]:
```suggestion
* Fix detection of macOS which was wrongly identified as Linux.
```

**Review by pellared — APPROVED** [review]

File: src/OpenTelemetry.Resources.OperatingSystem/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+## 0.1.0-alpha.2
+
+Released 2024-Jul-17
```
Comment (lachmatt) [inline src/OpenTelemetry.Resources.OperatingSystem/CHANGELOG.md:7]: These are added [automatically](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1950) by the prepare release workflow.

File: src/OpenTelemetry.Resources.OperatingSystem/OperatingSystemDetector.cs [hunk 4]
```diff
@@ -31,22 +34,13 @@ public Resource Detect()
...
-        if (platform == PlatformID.Unix)
-        {
-            return OperatingSystemsValues.Linux;
-        }
-
-        return null;
+#if NETFRAMEWORK
+        return OperatingSystemsValues.Windows;
+#else
+        return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? OperatingSystemsValues.Windows :
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.Resources.OperatingSystem/OperatingSystemDetector.cs:40]: nit: switch to if / switch for easier readability.

**Review by lachmatt — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Pull request merged**